mlflow/store/sqlalchemy_store.py

```diff
@@ -120,18 +115,21 @@ def _get_artifact_location(self, experiment_id):
     def create_experiment(self, name, artifact_location=None):
         if name is None or name == '':
             raise MlflowException('Invalid experiment name', INVALID_PARAMETER_VALUE)
+
+        new_session = self.SessionMaker()
         try:
             experiment = SqlExperiment(
                 name=name, lifecycle_stage=LifecycleStage.ACTIVE,
                 artifact_location=artifact_location
             )
-            self._save_to_db(experiment)
+            new_session.add(experiment)
             if not artifact_location:
                 # this requires a double write. The first one to generate an autoincrement-ed ID
-                experiment.artifact_location = self._get_artifact_location(experiment.experiment_id)
-                self._save_to_db(experiment)
+                eid = new_session.query(SqlExperiment).filter_by(name=name).first().experiment_id
+                experiment.artifact_location = self._get_artifact_location(eid)
+            new_session.commit()
```

Comment:
This is magic! Cool!

Reply:
Your idea! Weird that it works ... but then again, using local session for this was needed.